### PR TITLE
[IOTDB-500] Let timeColumn and batchData store time in the same struct

### PR DESCRIPTION
**Pull request by liutaohua**

### CODE REVIEW

File: tsfile/src/main/java/org/apache/iotdb/tsfile/read/common/BatchData.java [hunk 1]
```diff
@@ -21,6 +21,7 @@
 import java.io.Serializable;
 import java.util.ArrayList;
 import org.apache.iotdb.tsfile.common.conf.TSFileConfig;
```
Comment (samperson1997) [inline tsfile/src/main/java/org/apache/iotdb/tsfile/read/common/BatchData.java]: Remove this useless import

File: tsfile/src/main/java/org/apache/iotdb/tsfile/read/common/TimeColumn.java [hunk 2]
```diff
@@ -19,60 +19,115 @@
...
+  private int writeCurListIndex;
+  // inner array index for write
+  private int writeCurArrayIndex;
 
-  private int size;
+  // the insert timestamp number of timeRet
+  private int count;
 
-  private int cur;
+  private ArrayList<long[]> timeRet;
```
Comment (samperson1997) [inline tsfile/src/main/java/org/apache/iotdb/tsfile/read/common/TimeColumn.java]: How about using `List` here

File: tsfile/src/main/java/org/apache/iotdb/tsfile/read/common/TimeColumn.java [hunk 3]
```diff
@@ -19,60 +19,115 @@
...
+    this.writeCurArrayIndex = 0;
+    timeRet = new ArrayList<>();
+    timeRet.add(new long[capacity]);
+    count = 0;
   }
 
-  public TimeColumn(int initSize) {
-    times = new long[initSize];
-  }
+  public TimeColumn(ArrayList<long[]> timeRet, int count) {
```
Comment (samperson1997) [inline tsfile/src/main/java/org/apache/iotdb/tsfile/read/common/TimeColumn.java]: Also use `List`

File: tsfile/src/main/java/org/apache/iotdb/tsfile/read/common/TimeColumn.java [hunk 4]
```diff
@@ -19,60 +19,115 @@
...
 
   public boolean hasCurrent() {
-    return size > 0 && cur < size;
+    if (readCurListIndex < writeCurListIndex) {
+      return readCurArrayIndex < capacity;
+    } else if (readCurListIndex == writeCurListIndex) {
+      return readCurArrayIndex < writeCurArrayIndex;
+    } else {
+      return false;
+    }
```
Comment (samperson1997) [inline tsfile/src/main/java/org/apache/iotdb/tsfile/read/common/TimeColumn.java]: could be simpliy:
```suggestion
    } else {
      return readCurListIndex == writeCurListIndex && readCurArrayIndex < writeCurArrayIndex;
    }
```

File: tsfile/src/main/java/org/apache/iotdb/tsfile/read/common/TimeColumn.java [hunk 5]
```diff
@@ -19,60 +19,115 @@
...
 
-  public int size() {
-    return size;
+  public TimeColumnR asReadOnlyTimeColumn() {
+    return new TimeColumnR(timeRet, count);
+  }
+
+  private class TimeColumnR extends TimeColumn {
+
+    public TimeColumnR(ArrayList<long[]> timeRet, int count) {
```
Comment (samperson1997) [inline tsfile/src/main/java/org/apache/iotdb/tsfile/read/common/TimeColumn.java]: use `List` ?

File: tsfile/src/main/java/org/apache/iotdb/tsfile/read/common/BatchData.java (hunk 1 above)
Comment (liutaohua) [reply tsfile/src/main/java/org/apache/iotdb/tsfile/read/common/BatchData.java]: done

File: tsfile/src/main/java/org/apache/iotdb/tsfile/read/common/TimeColumn.java (hunk 2 above)
Comment (liutaohua) [reply tsfile/src/main/java/org/apache/iotdb/tsfile/read/common/TimeColumn.java]: done

File: tsfile/src/main/java/org/apache/iotdb/tsfile/read/common/TimeColumn.java (hunk 3 above)
Comment (liutaohua) [reply tsfile/src/main/java/org/apache/iotdb/tsfile/read/common/TimeColumn.java]: done

File: tsfile/src/main/java/org/apache/iotdb/tsfile/read/common/TimeColumn.java (hunk 5 above)
Comment (liutaohua) [reply tsfile/src/main/java/org/apache/iotdb/tsfile/read/common/TimeColumn.java]: done

File: tsfile/src/main/java/org/apache/iotdb/tsfile/read/common/TimeColumn.java (hunk 4 above)
Comment (liutaohua) [reply tsfile/src/main/java/org/apache/iotdb/tsfile/read/common/TimeColumn.java]: excellent advice,  exchange order will be more easy to understand

**Pull request merged**